### PR TITLE
[Components][EventDispatcher] don't explain deprecated `getName()` method

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Doc fix? | yes |
| New docs? | no |
| Applies to | 2.5+ |
| Fixed tickets |  |

### CODE REVIEW

**Comment by xabbuh** [comment]
I think it's better to explain the way to got in the earliest version this is available instead of simply removing the method in the `master` branch.

File: components/event_dispatcher/introduction.rst [hunk 1]
```diff
@@ -626,22 +626,21 @@ and so on...
...
-:class:`Symfony\\Component\\EventDispatcher\\Event` objects, making it available
-to event listeners via the :method:`Symfony\\Component\\EventDispatcher\\Event::getName`
-method.
+.. versionadded:: 2.4
+    Before Symfony 2.4, the event name and the event dispatcher had to be
+    requested from the ``Event`` instance. These methods are now deprecated.
 
-The event name, (as with any other data in a custom event object) can be used as
-part of the listener's processing logic::
+The `EventDispatcher` instance as well as the name of the dispatched event
```
Comment (WouterJ) [inline components/event_dispatcher/introduction.rst]: Double backticks :)

Comment (xabbuh) [inline components/event_dispatcher/introduction.rst]: oops, fixed

**Comment by WouterJ** [comment]
:+1:

File: components/event_dispatcher/introduction.rst [hunk 2]
```diff
@@ -626,22 +626,21 @@ and so on...
...
+The ``EventDispatcher`` instance as well as the name of the dispatched event
+itself, are passed as arguments to the listener::
 
     use Symfony\Component\EventDispatcher\Event;
+    use Symfony\Component\EventDispatcher\EventDispatcher;
 
     class Foo
     {
-        public function myEventListener(Event $event)
+        public function myEventListener(Event $event, $eventName, EventDispatcher $dispatcher)
```
Comment (stof) [inline components/event_dispatcher/introduction.rst]: the typehint should be the interface, not the implementation

Comment (xabbuh) [inline components/event_dispatcher/introduction.rst]: indeed, thanks

File: components/event_dispatcher/introduction.rst [hunk 3]
```diff
@@ -626,22 +626,21 @@ and so on...
...
-to event listeners via the :method:`Symfony\\Component\\EventDispatcher\\Event::getName`
-method.
+.. versionadded:: 2.4
+    Before Symfony 2.4, the event name and the event dispatcher had to be
+    requested from the ``Event`` instance. These methods are now deprecated.
 
-The event name, (as with any other data in a custom event object) can be used as
-part of the listener's processing logic::
+The ``EventDispatcher`` instance as well as the name of the dispatched event
+itself, are passed as arguments to the listener::
```
Comment (WouterJ) [inline components/event_dispatcher/introduction.rst]: The comma should be removed imo (or you should put "as well as [...] event itself" between comma's.

File: components/event_dispatcher/introduction.rst [hunk 4]
```diff
@@ -626,22 +626,21 @@ and so on...
...
-:class:`Symfony\\Component\\EventDispatcher\\Event` objects, making it available
-to event listeners via the :method:`Symfony\\Component\\EventDispatcher\\Event::getName`
-method.
+.. versionadded:: 2.4
+    Before Symfony 2.4, the event name and the event dispatcher had to be
+    requested from the ``Event`` instance. These methods are now deprecated.
 
-The event name, (as with any other data in a custom event object) can be used as
-part of the listener's processing logic::
+The ``EventDispatcher`` instance as well as the name of the dispatched event
```
Comment (WouterJ) [inline components/event_dispatcher/introduction.rst]: I had to think twice about the meaning of "the dispatched event". What do you think about using "the event that is dispatched"?

**Comment by xabbuh** [comment]
@WouterJ Good suggestions, I changed the sentence. Thanks!

**Comment by weaverryan** [comment]
Well done! :thumbsup:

**Pull request merged**